numpy/core/tests/test_numeric.py

```diff
@@ -1446,6 +1446,19 @@ def test_array_equal(self):
         assert_(res)
         assert_(type(res) is bool)
 
+    def test_array_equal_equal_nan(self):
+        # Test array_equal with equal_nan kwarg
+        a1 = np.array([1, 2, np.nan])
+        a2 = np.array([1, np.nan, 2])
+        a3 = np.array([1, 2, np.inf])
+
+        # equal_nan=False by default for backwards compatibility
```

Comment:
I disagree :), `equal_nan` is the correct default in any case.

Reply:
Haha good point - I've let a little too much of my own personal perspective leak into the comment :)